apps/storefront/src/gatsby-theme-docz/components/MainContainer/index.js

```diff
@@ -110,15 +111,62 @@ const LandingPage = styled.div``
 
 export const MainContainer = ({ children, doc, ...rest }) => {
   const {
-    tabs,
+    tabs: rawTabs,
     title,
     toc,
     route,
     mode = 'draft',
     type = 'contentPage',
     slug,
   } = doc.value
-  const withTabs = tabs !== undefined
+
+  const docs = useDocs()
+
+  const drafts = useMemo(
+    () =>
+      R.pipe(
+        R.map(R.pick(['route', 'mode'])),
+        R.filter((doc) => doc.mode.toLowerCase() === 'draft'),
+      )(docs),
+    [docs],
+  )
+
+  const findByRoute = (collection) => (route) => {
+    return R.find(R.propEq('route', route), collection)
+  }
+
+  const isDraft = findByRoute(drafts)
+
+  const reduceIndexed = R.addIndex(R.reduce)
+
+  const publishedTabs = (drafts) => (route) => (tabs) =>
+    R.pipe(
+      reduceIndexed(
+        (acc, curr, i) => [
+          ...acc,
+          {
+            label: curr,
+            route:
+              route.match(/^\/([a-z-]+\/?){2}/)[0] +
+              (i === 0
+                ? ''
+                : R.pipe(R.toLower, R.curry(R.replace)(/\s/g, '-'))(curr) +
+                  '/'),
+          },
+        ],
+        [],
+      ),
```

Comment:
🤦

Reply:
Not particularly pleased with how that publishedTabs function looks.

Reply:
@vnys It looks like you are using the index to ignore the first entry in the list. Maybe you can use the [tail](https://ramdajs.com/docs/#tail) function instead? Would be something like this instead, `pipe(tail, map, filter)`

Reply:
Actually I don’t ignore the first entry in the list, I just do something different to the route. Not sure how I would go about doing something like that with Ramda. But i replaced reduce with map, makes it a bit easier to read.